pages/api/studentPortal/quizzes/scoringLogicFPL.ts

```diff
@@ -0,0 +1,21 @@
+import { QuizViewState } from "../../../../components/resources/quizzes/shared/SkillSelections";
+
+export const computeScore = (quizViewState: QuizViewState) => {
```

Comment:
good use of reduce, makes sense to me to use this iterator. is it possible to simplify this and get away with just one reduce? not sure if its possible just curious

Reply:
this is hard to read, ask chatgpt to rewrite it for you to make it more readable. Add unit tests for this function in this PR. 

If you still can't think of a cleaner way to write this, I'll try to help.

Reply:
`const score = { JavaScript: 0, "HTML/CSS": 0, Python: 0 };
  
  quizViewState.questions.map((question) => {
    question.options.map((option) => {
      if (option.isSelected && option.result) {
        score[option.result] += option.weight;
      }
    });
  });
  return score;`

Reply:
Come up with a simplified solution!

Reply:
excellent work Jimi! Super clean code, and makes my job implementing similar career-quiz results logic so much easier!

Reply:
you're new solution looks good, lets add a unit test for this function